lib/task/import/csvAuthorityRecordImportTask.class.php

```diff
@@ -535,29 +535,53 @@ public function execute($arguments = array(), $options = array())
               }
               else
               {
-                $relation = new QubitRelation;
-                $relation->subjectId = $sourceActorId;
-                $relation->objectId  = $targetActorId;
-                $relation->typeId    = $relationTypeId;
-
-                if ($self->rowStatusVars['date'])
-                {
-                  $relation->date = $self->rowStatusVars['date'];
-                }
-                if ($self->rowStatusVars['startDate'])
-                {
-                  $relation->startDate = $self->rowStatusVars['startDate'];
-                }
-                if ($self->rowStatusVars['endDate'])
+                // Check for pre-existing relationship before adding
+                $sql = "SELECT id FROM relation \r
+                  WHERE subject_id=:subject_id \r
+                  AND object_id=:object_id \r
+                  AND type_id = :type_id";
+
+                $params = array(
+                  ':subject_id' => $sourceActorId,
+                  ':object_id' => $targetActorId,
+                  ':type_id' => $relationTypeId
+                );
+
+                $paramsVariant = array(
+                  ':subject_id' => $targetActorId,
+                  ':object_id' => $sourceActorId,
+                  ':type_id' => $relationTypeId
+                );
```

Comment:
Does it matter that this relation is the opposite of the one being created? Or is this the source of the duped relations?

Reply:
I think a relationship can be specified in two ways, so I'm checking for both.